wallet/wallet-controller/src/lib.rs

```diff
@@ -376,12 +375,38 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static, W: WalletEvents> Controll
             .map_err(ControllerError::WalletError)
     }
 
+    /// Broadcast a singed transaction to the mempool and update the wallets state if the
+    /// transaction has been added to the mempool
+    async fn broadcast_to_mempool(
+        &mut self,
+        tx: SignedTransaction,
+    ) -> Result<TxStatus, ControllerError<T>> {
+        let status = self
+            .rpc_client
+            .submit_transaction(tx.clone())
+            .await
+            .map_err(ControllerError::NodeCallError)?;
+
+        match status {
+            mempool::TxStatus::InMempool => {
+                self.wallet
+                    .add_unconfirmed_tx(tx, &mut self.wallet_events)
+                    .map_err(ControllerError::WalletError)?;
+            }
+            mempool::TxStatus::InOrphanPool => {
+                // Mempool should reject the transaction and not return `InOrphanPool`
```

Comment:
Please write an error log entry about this so that if it happens we know

Reply:
Mempool will currently happily add transaction to orphan pool. There's an issue for it #1082

Reply:
added a log and a comment, that we will ignore such transactions to prevent building upon them and getting stuck.